lib/cdo/firehose.rb

```diff
@@ -34,20 +31,39 @@ def initialize
   # @param stream_name [String] The steam to put the data in.
   # @param data [hash] The data to insert into the stream.
   def put_record(stream_name, data)
+    data_with_common_values = add_common_values(data)
+
     # TODO(asher): Determine whether these should be cached and batched via
     # put_record_batch. See
     #   http://docs.aws.amazon.com/sdkforruby/api/Aws/Firehose/Client.html#put_record_batch-instance_method
     # for documentation.
     @firehose.put_record(
       {
         delivery_stream_name: stream_name,
-        record: {data: data.to_json}
+        record: {data: data_with_common_values.to_json}
       }
     )
-  rescue Aws::Firehose::Errors::ServiceError
+  rescue Aws::Firehose::Errors::ServiceError => e
     # TODO(asher): Determine what action is appropriate here. In particular,
     # if the exception is ServiceUnavailableException, we should consider
     # backing off and retrying.
     # See http://docs.aws.amazon.com/sdkforruby/api/Aws/Firehose/Client.html#put_record-instance_method.
+    Honeybadger.notify(e)
+  end
+
+  private
+
+  # Adds common key-value pairs to the data hash.
+  # @param data [hash] The data to add the key-value pairs to.
+  # @return [hash] The data, including the newly added key-value pairs.
+  def add_common_values(data)
+    data_with_common_values = data.merge(
+      created_at: DateTime.now,
+      environment: rack_env,
+      device: 'server-side'.to_json
```

Comment:
Perhaps? For now, I'm going to stick with the generic "server-side", though this is definitely something we should consider in the future.